pokt-network/pocket
# V1 P2P Prototype

### Overview
* #### Issue: [Issue#25](https://github.com/pokt-network/pocket/issues/22)
* #### Guidelines for merging:
  

- [ ] make sure #27 is merged first
- [ ] Pull and rebase and fix conflicts
- [ ] Change the base to `milestone/v1-prototype`

* #### Impact:
    * Added a new module: P2P prototype
    * Updated the existing p2p mock to use P2P protobuf messages
    * Deprecated unused/unnecessary p2p types used by the p2p mock
* #### Description:

This PR adds in the p2p prototype on top of the v1 snapshot as in, and keeps in place the existing p2p mock (_at least for the time being_)

* #### Testing:

**_Prerequisite_**: The development environment and instructions [here](https://drive.google.com/file/d/1Uc43PQ4t15GyT5t1bpsv_KpCwyu9HxBa) are not expected to be complete since the development environment is still a work in progress, but the two following assumptions are made:

- Docker is installed and the daemon is running locally.
- protoc is installed and running locally.

* Test the p2p module
```
$ go test -v -race ./p2p
```

* Run the e2e script
```
$ E2E=true go test -v -run TestNetwork_E2E -v -race ./p2p
```

* Run the e2e stack
```
make p2p-run-e2e-stack
```

* Test the p2p integration with the rest of the modules:

Shell#1
```
$ make compose_and_watch
```

Shell#2
```zsh
$ USE_P2P_MOCK=false make client_start && make client_connect
```
After you are prompted with the choices, select whatever action you see fit.
